File: Stars Pattern/Butterfly/Pattern.py
#!/usr/bin/env python3

"""
Butterfly Pattern
"""

from sys import path

path.append("../.")


from ConsoleNumInput import inInt


#### Begin ####

n = inInt().getInput(5)

# Method 1
print("Method 1")

# Upper Part 
for x in range(1, n): 
    for y in range(x):
        print(" * ", end="")
            
    for y in range(((n * 2) - (x * 2)) - 1):
        print("   ", end="")
    
    for y in range(x):
        print(" * ", end="")
    
    print("")

# Middle Part
for x in range((n * 2) - 1):
    print(" * ", end="")

print("")

# Lower Part. Printed in Reverse
for x in range(n - 1, -1, -1): 
    for y in range(x):
        print(" * ", end="")
            
    for y in range(((n * 2) - (x * 2)) - 1):
        print("   ", end="")
    
    for y in range(x):
        print(" * ", end="")
    
    print("")


print("\n*~*~*~*~*~*~*~*~*~*~*~*\n")


# Method 2

print("Method 2")

for x in range(1, n):
    
    print(" * " * x, end="")
    print("   " * ((n * 2) - (x * 2) - 1), end="")
    print(" * " * x)

print(" * " * ((n * 2) - 1))

for x in range(n - 1, -1, - 1):
    
    print(" * " * x, end="")
    print("   " * ((n * 2) - (x * 2) - 1), end="")
    print(" * " * x)


print("")
"""
Author: Jovan De Guia
Github Username: jxmked
"""

File: Stars Pattern/Butterfly/Pattern.c
#include <stdio.h>
#include <stdlib.h>

// External File
#include "../ConsoleNumInput.c"

/**
 * Butterfly Pattern
 * */

static int loop(int n);


int main(void){
    // Default
    int n = 5;
    
    do {
        // From External File
        clrscr();
        
        loop(n);
        
        n = inInt();
        
        // This will never trigger. 
        // System required. Hahaha
        if(! n){
            return 0;
        }
    } while(1);
}

static int loop(int n){
    int i, j;
    
    for(i = 0; i < n; i++){
        for(j = 0; j < (i + 1); j++){
            printf(" * ");
        }
        
        for(j = (i * 2); j < ((n * 2) - 1); j++){
            printf("   ");
        }
        
        for(j = 0; j < (i + 1); j++){
            printf(" * ");
        }
        
        printf("\n");
    }
    
    for(i = 0; i < ((n * 2) + 1); i++){
        printf(" * ");
    }
    
    printf("\n");
    
    for(i = n - 1; i >= 0; i--){
        for(j = 0; j < (i + 1); j++){
            printf(" * ");
        }
        
        for(j = (i * 2); j < ((n * 2) - 1); j++){
            printf("   ");
        }
        
        for(j = 0; j < (i + 1); j++){
            printf(" * ");
        }
        
        printf("\n");
    }
    
    return 0;
}

/**
 * Written by Jovan De Guia
 * Github Username: jxmked
 * */
